src/auth/auth.service.ts

```diff
@@ -0,0 +1,49 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { JwtService } from '@nestjs/jwt';
+import { AuthDto } from './dto/auth.dto';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService,
+  ) {}
+
+  async signup(dto: AuthDto) {
+    const hash = await bcrypt.hash(dto.password, 10);
+
+    const user = await this.prisma.user.create({
+      data: {
+        email: dto.email,
+        username: dto.username,
+        password: hash,
+      },
+    });
+
+    return this.signToken(user.id, user.email);
+  }
+
+  async login(dto: AuthDto) {
+    const user = await this.prisma.user.findUnique({
+      where: { email: dto.email },
+    });
+
+    if (!user || !(await bcrypt.compare(dto.password, user.password))) {
+      throw new ForbiddenException('Invalid credentials');
+    }
+
+    return this.signToken(user.id, user.email);
+  }
+
+  async signToken(userId: number, email: string): Promise<{ access_token: string }> {
```

Comment:
```suggestion
  private async signToken(userId: number, email: string): Promise<{ access_token: string }> {
```
method này chỉ dùng trong nội bộ class,
nên set scope `private` cho nó